xarray/indexes/__init__.py

```diff
@@ -4,10 +4,17 @@
 """
 
 from xarray.core.indexes import (
+    CoordinateTransformIndex,
```

Comment:
@benbovy does this seem OK to you? Would be good for SciPy...

Reply:
Yes! We should also expose `xarray.core.coordinate_transform.CoordinateTransform` since we already subclass it externally (not sure in which namespace... `indexes` as well?).